implement py-person-class

Tario1 opened the pull request.

Tario1 committed:
implement py-person-class

mateacademy-ai-mentor commented on app/main.py:
```diff
@@ -1,8 +1,27 @@
+    persons_list = [Person(person_dict["name"],
+                    person_dict["age"]) for person_dict
+                    in people if person_dict.get("name")
+                    and person_dict.get("age") is not None]
```
This list comprehension uses direct dictionary indexing (`person_dict["name"]` and `person_dict["age"]`) which can raise KeyError and does not follow the checklist guidance. This violates checklist item #4: "When accessing values in a dictionary, it is better to use the `dict.get()` method instead of explicitly checking for the presence of a key. This approach simplifies the code and avoids potential `KeyError` exceptions, making it cleaner and more concise." Please use `person_dict.get('name')` and `person_dict.get('age')` (and keep the existence guard) so dictionary access follows the checklist recommendation. Example: `Person(person_dict.get('name'), person_dict.get('age'))` while ensuring your guard still prevents creating Person for missing values. See checklist for the exact wording .

mateacademy-ai-mentor commented on app/main.py:
```diff
@@ -1,8 +1,27 @@
 def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+    Person.people = {}
```
The function is declared as `def create_person_list(people: list) -> list:` but the task description requires the function signature to be `create_person_list(people: list[dict]) -> list[Person]`. Please update the type annotations to match the requirement from the description so the signature clearly documents the expected input and output types. Reference: the description specifies `Implement a function create_person_list(people: list[dict]) -> list[Person]:` .

mateacademy-ai-mentor reviewed (changes requested):
### Review Summary

I am requesting changes because two checklist/requirement-aligned issues remain that could cause errors and do not meet the specified signature.

Critical issues to fix:
1. In app/main.py, within the list comprehension that builds `Person` instances (around the lines where `Person(person_dict["name"], person_dict["age"])` is used), you directly index the dict with `person_dict["name"]` and `person_dict["age"]`. This can raise `KeyError` and violates the requirement to prefer `dict.get()` [CHECKLIST ITEM #4]. Update to use `person_dict.get("name")` and `person_dict.get("age")`, and keep your guard so you skip entries missing these values (e.g., check `person_dict.get("name") is not None and person_dict.get("age") is not None`).
2. The function signature currently is `def create_person_list(people: list) -> list:`. The task requires `def create_person_list(people: list[dict]) -> list[Person]:`. Update the type hints to match exactly so the input and output types are explicit and compliant with the description.

Minor improvement (optional): If empty-string names should be allowed, adjust the guard to check for `is not None` rather than truthiness. 

You did a solid job on using a list comprehension, populating `Person.people` in `__init__`, and linking spouses via object references. Make these tweaks and you’ll be all set!

***

✨ Please rate this review in the Platform's notification center[^1]. I will be grateful for your feedback! ✨

[^1]: [Rate AI review example](https://snipboard.io/gD6XKF.jpg)

> [!IMPORTANT]
> Fix the issues, commit the changes, and then re-request my review using the **`Re-request review` button**[^2] in the Reviewers tab.

[^2]: [Re-request AI review example](https://snipboard.io/nOXxHN.jpg)

> [!TIP]
> If you have any comments about this review, please leave a message in the [Add a comment](#new_comment_field) section below and **mention me with the `@mateacademy-ai-mentor` tag**[^3].
> I will read your message during my next review of your pull request. Please do not reply directly to my comments, as that feature is not supported yet.

[^3]: [Add AI review comments example](https://snipboard.io/hVlOey.jpg)

> [!CAUTION]
> Do not close the pull request until the review is complete. Closing it early can lead to unexpected behavior and disrupt the review process.

Tario1 committed:
implement py-person-class fixes

added def create_person_list(people: list[dict]) -> list[Person]

mateacademy-ai-mentor reviewed (commented):
Sorry, the pull request review was unsuccessful for the following reasons:

> [!WARNING]
> **No changes were made in the pull request since the last review.** Please make sure that you have added files to the pull request and committed the changes.
> <sub>*[CODE: NCM]*</sub>


If the issue is within your control, please address it.
The next review attempt <ins>will automatically launch</ins> in about 1 hour.
If you believe this is an error or need assistance, please contact the support team.

mateacademy-ai-mentor commented on app/main.py:
```diff
@@ -1,8 +1,27 @@
+    persons_list = [Person(person_dict["name"],
+                    person_dict["age"]) for person_dict
+                    in people if person_dict.get("name")
+                    and person_dict.get("age") is not None]
```
This violates checklist item #4: "When accessing values in a dictionary, it is better to use the `dict.get()` method instead of explicitly checking for the presence of a key. This approach simplifies the code and avoids potential `KeyError` exceptions, making it cleaner and more concise." 

Concretely, the list comprehension constructs Person instances using direct indexing `person_dict["name"]` and `person_dict["age"]` on these lines in app/main.py: replace those direct indexes with `person_dict.get("name")` and `person_dict.get("age")`. Also keep/adjust the guard so you only create instances when `person_dict.get("name") is not None` and `person_dict.get("age") is not None` to avoid creating invalid Person objects. Refer to the current code around these lines: app/main.py lines 12-15.

mateacademy-ai-mentor reviewed (changes requested):
### Review Summary

I’m requesting changes because there’s a critical checklist violation that could lead to runtime errors. In app/main.py lines 12–15, the list comprehension constructs `Person` with `person_dict["name"]` and `person_dict["age"]`, which can raise `KeyError` and violates [CHECKLIST ITEM #4]. Please replace direct indexing with `person_dict.get("name")` and `person_dict.get("age")`, and keep/adjust the guard so instances are only created when both values are not `None` (e.g., `if person_dict.get("name") is not None and person_dict.get("age") is not None`).

What you did well: the `Person` class, class-level registry, two-pass creation then spouse-linking, and correct type hints are all solid. Minor improvement: if empty strings for names should be allowed, avoid truthiness checks and use `is not None` explicitly. Once you make the `dict.get()` change and guard correctly, this will meet the checklist and I can approve. Keep up the good work!

***

✨ Please rate this review in the Platform's notification center[^1]. I will be grateful for your feedback! ✨

[^1]: [Rate AI review example](https://snipboard.io/gD6XKF.jpg)

> [!IMPORTANT]
> Fix the issues, commit the changes, and then re-request my review using the **`Re-request review` button**[^2] in the Reviewers tab.

[^2]: [Re-request AI review example](https://snipboard.io/nOXxHN.jpg)

> [!TIP]
> If you have any comments about this review, please leave a message in the [Add a comment](#new_comment_field) section below and **mention me with the `@mateacademy-ai-mentor` tag**[^3].
> I will read your message during my next review of your pull request. Please do not reply directly to my comments, as that feature is not supported yet.

[^3]: [Add AI review comments example](https://snipboard.io/hVlOey.jpg)

> [!CAUTION]
> Do not close the pull request until the review is complete. Closing it early can lead to unexpected behavior and disrupt the review process.

Tario1 committed:
implement py-person-class fixes (now it should go trough all tests)

[Person(person_dict("name"),person_dict("age")) to [Person(person_dict.get("name"),person_dict.get("age"))
changed lines:      persons_list = [Person(person_dict.get("name"),
                    person_dict.get("age")) for person_dict
                    in people if person_dict.get("name") is not None
                    and person_dict.get("age") is not None]

mateacademy-ai-mentor reviewed (approved):
### Review Summary

Your solution is approved. You implemented the Person class with the exact `__init__(name, age)` signature, maintained a class-level `people` registry, created instances via list comprehension, and correctly performed spouse linking in a second pass using `dict.get()`—all aligning with the checklist and requirements.

- Potential improvements (educational override to help you progress while learning):
  - At app/main.py line where spouse linking occurs (the `if partner_name:` guard), consider checking `partner_name is not None` instead of truthiness so empty-string names aren’t skipped. This ensures links aren’t missed when a valid name is an empty string. [CHECKLIST ITEM: avoid brittle truthy checks]
  - Near the `__init__`, consider validating `age` is an `int` to prevent accidental non-integer values being stored.
  - In `create_person_list`, you reset `Person.people` globally; document this side effect or avoid resetting if a shared registry across calls is desired.

Overall, strong adherence to the two-pass creation/linking pattern, clean use of `dict.get()`, and avoiding nested conditionals. Nice work—keep it up!

***

✨ Please rate this review in the Platform's notification center[^1]. I will be grateful for your feedback! ✨

[^1]: [Rate AI review example](https://snipboard.io/gD6XKF.jpg)

piotrzegarek commented on app/main.py:
```diff
@@ -1,8 +1,27 @@
+                    in people if person_dict.get("name") is not None
+                    and person_dict.get("age") is not None]
+
+    for person_dict in people:
+        current_person = Person.people.get(person_dict.get("name"))
+        if not current_person:
+            continue
+        partner_name = person_dict.get("wife")
+        if partner_name and partner_name in Person.people:
+            current_person.wife = Person.people[partner_name]
+        partner_name = person_dict.get("husband")
+        if partner_name and partner_name in Person.people:
+            current_person.husband = Person.people[partner_name]
+    return persons_list
```
Logika poprawna chociaż można to mocno uprościć używając funkcji `setattr()`, spróbuj to poprawić.

Tario1 committed:
Changed this lines

        partner_name = person_dict.get("wife")
        if partner_name and partner_name in Person.people:
            current_person.wife = Person.people[partner_name]
        partner_name = person_dict.get("husband")
        if partner_name and partner_name in Person.people:
            current_person.husband = Person.people[partner_name]

to this lines to use setattr as advised by mentor (had to look around for that) and it really looks cleaner and code is not repeated for husband and wife

        for relation in ("wife", "husband"):
            partner_name = person_dict.get(relation)
            partner = Person.people.get(partner_name)
            if partner:
                setattr(current_person, relation, partner)

Czernich reviewed (approved):
Looks good to me
